File: Python/Skills.py
from Abilities import Abilities

class Skills(Abilities):
    """Kemampuan yang menggunakan Action Point (AP)."""
    def __init__(self, name: str, description: str, type: str, cooldown: int, apCost: int):
        super().__init__(name, description, type, cooldown)
        self.apCost = apCost

    def display(self):
        super().display()
        print(f"    Biaya AP: {self.apCost}")

File: C++/Skills.cpp
#pragma once
#include "Abilities.cpp"

class Skills : public Abilities {
private:
    int apCost;

public:
    Skills(std::string name, std::string description, std::string type, int cooldown, int apCost)
        : Abilities(name, description, type, cooldown), apCost(apCost) {}

    // Getter
    int getApCost() const { return apCost; }

    // Setter
    void setApCost(int ap) { apCost = ap; }

    // Override Display
    void display() const override {
        Abilities::display();
        std::cout << "    Biaya AP: " << apCost << std::endl;
    }
};